databroker/src/grpc/kuksa_val_v1/val.rs

```diff
@@ -72,7 +75,9 @@ impl proto::val_server::Val for broker::DataBroker {
 
             // Fill valid_requests structure.
             for request in requested {
-                if request.path.contains('*') && !glob::is_valid_pattern(&request.path) {
+                if (request.path.contains('*') && !glob::is_valid_pattern(&request.path))
+                    || request.path.len() > MAX_REQUEST_PATH_LENGTH
```

Comment:
Done

Reply:
Doesn't it make more sense to check for a valid pattern when creating the glob matcher, i.e. the proposed `glob::Matcher`, or if we're not doing that, when creating the "glob string".

Reply:
`is_valid_pattern` signature I think is not correct, would be better if `is_valid_path`.
Here it should check if path requested by the user is valid, not if the glob string is valid.

Reply:
Since we use request paths with point separators and asterisks in wildcards for all the kuksa clients would not be enough for now to check just if the request path is a valid request pattern?.
Otherwise I have to implement a new glob string regex (last time took me long time). 
`is_valid_pattern` as requested path "filter" for now is probably enough?

Reply:
> Since we use request paths with point separators and asterisks in wildcards for all the kuksa clients would not be enough for now to check just if the request path is a valid request pattern?

I don't think I understand what you mean.. And I guess I was a bit unclear as well.

What I'm saying is:

1. Why is the path only validated if the path contains a `*`. Shouldn't the path always be validated?
2. Why is `glob::is_valid_pattern` not part of `glob::Matcher::new()`. It should not be possible to create a glob matcher with an invalid path, so it should be part of the constructor. Right now, the user has to remember to do the validation. It's better to have the Rust type system guarantee this.

>  Otherwise I have to implement a new glob string regex (last time took me long time).

Why? The validation function is already there.

Reply:
Got it now, introduced the Matcher struct that wraps that functionality, you can take a look at it